src/adagrams.js

```diff
@@ -1,15 +1,131 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+    const LETTER_POOL = {
+      'A': 9, 
+      'B': 2, 
+      'C': 2, 
+      'D': 4, 
+      'E': 12, 
+      'F': 2, 
+      'G': 3, 
+      'H': 2, 
+      'I': 9, 
+      'J': 1, 
+      'K': 1, 
+      'L': 4, 
+      'M': 2, 
+      'N': 6, 
+      'O': 8, 
+      'P': 2, 
+      'Q': 1, 
+      'R': 6, 
+      'S': 4, 
+      'T': 6, 
+      'U': 4, 
+      'V': 2, 
+      'W': 2, 
+      'X': 1, 
+      'Y': 2, 
+      'Z': 1
+      };
+
+    const alphabet = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+
+    const hand = [];
+    let letterPool = JSON.parse(JSON.stringify(LETTER_POOL));
+    
+    while(hand.length < 10) {
+        const randomLetter = alphabet[Math.floor(Math.random() * alphabet.length)];
+        if (letterPool[randomLetter] > 0) {
+            letterPool[randomLetter] -= 1;
+            hand.push(randomLetter);
+        }
+    }
+
+    return hand;
 };
 
+
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
-};
+        for (let i in input) {
+            if (!(lettersInHand.includes(input[i]))) {
+                return false;
+            } else {
+                let index = lettersInHand.indexOf(input[i]);
+                lettersInHand.splice(index, 1);
```

Comment:
The time complexity of indexOf and splice in JS is the same as it is in Python, linear. How could you refactor this method to use an object so that when you look for a letter, you'll have constant time instead of linear?